workflow/scripts/osemosys_global/transmission/user_defined_capacity.py

```diff
@@ -0,0 +1,244 @@
+                
+        # For technologies with start year at or after model start year, add to 
+        # TotalAnnualMinCapacityInvestment      
+        df_min_cap_inv = df_min_cap_inv.loc[df_min_cap_inv['YEAR'] >= min(get_years(start_year, end_year))]
+        df_min_cap_inv.drop_duplicates(subset=['REGION', 
+                                               'TECHNOLOGY',
+                                               'YEAR'],
+                                       keep='last',
+                                       inplace=True)
+        
+        # Add IAR and OAR for custom technologies
+        tech_list = list(tech_capacity_trn_df['TECHNOLOGY'].unique())
+        df_iar_custom = pd.DataFrame(list(itertools.product(tech_list,
+                                                        [1, 2],
+                                                        get_years(start_year, end_year))
+                                      ),
+                                 columns = ['TECHNOLOGY',
+                                            'MODE_OF_OPERATION',
+                                            'YEAR']
+                                 )
+        df_oar_custom = pd.DataFrame(list(itertools.product(tech_list,
+                                                        [1, 2],
+                                                        get_years(start_year, end_year))
+                                      ),
+                                 columns = ['TECHNOLOGY',
+                                            'MODE_OF_OPERATION',
+                                            'YEAR']
+                                 )  
+        # IAR in modes 1 and 2 are primary electricity commodity ('ELC*01') in 
+        # node_from and node_to, respectively. 
+        # OAR is the inverse of the above
+        df_iar_custom.loc[df_iar_custom['MODE_OF_OPERATION']==1,
+                                        'FUEL'] = ('ELC' + 
+                                                   df_iar_custom['TECHNOLOGY'].str[3:8] + 
+                                                   '01')
+        df_iar_custom.loc[df_iar_custom['MODE_OF_OPERATION']==2,
+                                        'FUEL'] = ('ELC' + 
+                                                   df_iar_custom['TECHNOLOGY'].str[8:13] + 
+                                                   '01')
+        df_oar_custom.loc[df_iar_custom['MODE_OF_OPERATION']==1,
+                                        'FUEL'] = ('ELC' + 
+                                                   df_oar_custom['TECHNOLOGY'].str[8:13] + 
+                                                   '01')
+        df_oar_custom.loc[df_iar_custom['MODE_OF_OPERATION']==2,
+                                        'FUEL'] = ('ELC' + 
+                                                   df_oar_custom['TECHNOLOGY'].str[3:8] + 
+                                                   '01')
```

Comment:
Are these correct for transmission techs? Dont the `01` and `02` identifiers change based on the mode of operation? 

![image](https://github.com/user-attachments/assets/1dc2a0e6-4e14-4f67-8069-51b84c47acfb)

Reply:
Correct and changed